### PR TITLE
Added GetSubscriptionStatsForPartitionedTopic method

### PR DESCRIPTION
**Pull request by vishwajitparasar-bsft**
Also fixed few typos with method names. Removed a deprecated method.

### CODE REVIEW

File: pulsarlib/api.go [hunk 1]
```diff
@@ -462,7 +462,7 @@ func CreatePartitionedTopic(tenantID string, namespace string, topic string, par
 	return nil
 }
 
-func DeletePartionedTopic(tenantID string, namespace string, topic string) error {
+func DeletePartitionedTopic(tenantID string, namespace string, topic string) error {
```
[the remark this answers is not in the document]
Comment (vishwajitparasar-bsft) [reply pulsarlib/api.go:465]: Yep we can mark it as a breaking change and bump it up to 1.0.0

File: pulsarlib/api.go [hunk 2]
```diff
@@ -712,3 +712,41 @@ func SetAutoSubscriptionCreationOnNamespace(tenantID, namespace string) error {
...
+	err = json.Unmarshal(respBuf, &result)
+	if err != nil {
+		return result, err
+	}
+
+	if result["subscriptions"] == nil {
+		return result, fmt.Errorf("failed to find subscriptions field in pulsar response")
+	}
+
+	subs := result["subscriptions"].(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (vishwajitparasar-bsft) [reply pulsarlib/api.go:744]: We could but also YAGNI 😆

[the remark this answers is not in the document]
Comment (sagarrohankar-bsft) [reply pulsarlib/api.go]: In this case, we are writing library, and you never know client will use it, giving more broader response will help client to chose what they want.

[the remark this answers is not in the document]
Comment (merajblueshift) [reply pulsarlib/api.go]: Agree with Sagar's comments. Let us expose API to return full response. Only thing is response should be well documented.

File: pulsarlib/api.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (merajblueshift) [reply pulsarlib/api.go:465]: Was any client using this function?

[the remark this answers is not in the document]
Comment (merajblueshift) [reply pulsarlib/api.go:465]: Clients would have anyways locked on a particular version. If they upgrade the compilation will fail and they will have to rename the call.

File: pulsarlib/api.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vishwajitparasar-bsft) [reply pulsarlib/api.go]: Done.

**Review by sagarrohankar-bsft — APPROVED** [review]
LG

**Review by merajblueshift — APPROVED** [review]

File: pulsarlib/api.go [hunk 3]
```diff
@@ -712,3 +712,30 @@ func SetAutoSubscriptionCreationOnNamespace(tenantID, namespace string) error {
...
+	if err != nil {
+		return nil, err
+	}
+
+	respBuf, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(respBuf, &result)
```
Comment (merajblueshift) [inline pulsarlib/api.go:735]: We should check resp.status_code before this.

**Pull request merged**